src/components/Nav/Nav.js

```diff
@@ -1,7 +1,165 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faMagnifyingGlass,
+  faCircleXmark,
+} from '@fortawesome/free-solid-svg-icons';
+import {
+  faBell,
+  faAddressBook,
+  faUser,
+} from '@fortawesome/free-regular-svg-icons';
+
+const BUTTON_LIST = [
+  { id: 1, icon: faBell },
+  { id: 2, icon: faAddressBook },
+  { id: 3, icon: faUser },
+];
 
 const Nav = () => {
-  return <div>This is Nav</div>;
+  return (
+    <NavSection>
+      <LoginSection>
+        <LoginButton>로그인</LoginButton>
+        <LoginButton>캐시충전</LoginButton>
+      </LoginSection>
+      <Navigator>
+        <TitleH1>
+          <LinkTag>
+            <DevUBlueTitle>DEVU</DevUBlueTitle>
+            <DevUBlackTitle>BOOKS</DevUBlackTitle>
+          </LinkTag>
+        </TitleH1>
+        <SearchLabel>
+          <MagnifyIcon icon={faMagnifyingGlass} />
+          <SearchInput />
+          <DeleteButton>
+            <DeleteIcon icon={faCircleXmark} />
+          </DeleteButton>
+        </SearchLabel>
+        <ButtonsSection>
+          {BUTTON_LIST.map(button => {
+            return (
+              <ButtonsWrap key={button.id}>
+                <ButtonsIcon icon={button.icon} />
+              </ButtonsWrap>
+            );
+          })}
+        </ButtonsSection>
+      </Navigator>
+    </NavSection>
+  );
 };
 
 export default Nav;
+
+const NavSection = styled.div`
+  max-width: 1170px;
+  margin: auto;
+  padding: 0 18px;
+`;
+
+const LoginSection = styled.div`
+  ${props => props.theme.variables.flex(null, 'flex-end')}
```

Comment:
함수의 매개변수 자리에서 구조분해할당 하실 수도 있습니다!
```suggestion
  ${({ theme }) => theme.variables.flex(null, 'flex-end')}
```

Reply:
헉.. 감사합니다!! 코드가 짧아지겠네요